packages/broker/src/config/ConfigWizard.ts

```diff
@@ -1,254 +1,539 @@
-        message: 'Please provide the private key to import',
-        when: (answers: inquirer.Answers): boolean => {
-            return answers.generateOrImportPrivateKey === PRIVATE_KEY_SOURCE_IMPORT
-        },
-        validate: (input: string): string | boolean => {
-            try {
-                new Wallet(input)
-                return true
-            } catch (e: any) {
-                return 'Invalid private key provided.'
-            }
-        }
-    },
-    {
-        type: 'confirm',
-        name: 'revealGeneratedPrivateKey',
-        // eslint-disable-next-line max-len
-        message: 'We strongly recommend backing up your private key. It will be written into the config file, but would you also like to see this sensitive information on screen now?',
-        default: false,
-        when: (answers: inquirer.Answers): boolean => {
-            return answers.generateOrImportPrivateKey === PRIVATE_KEY_SOURCE_GENERATE
+/**
+ * Lets the user decide where to write the config file.
+ */
+async function getStoragePath() {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+        const path = await input({
+            message: 'Select a path to store the generated config in',
+            default: getDefaultFile(),
+        })
+
+        const proceed =
+            !existsSync(path) ||
+            (await confirm({
+                message: `The selected destination ${path} already exists. Do you want to overwrite it?`,
+                default: false,
+            }))
+
+        if (proceed) {
+            return path
         }
     }
-]
+}
 
-const createPluginPrompts = (): Array<inquirer.Question | inquirer.ListQuestion | inquirer.CheckboxQuestion> => {
-    const selectApiPluginsPrompt: inquirer.CheckboxQuestion = {
-        type: 'checkbox',
-        name: 'enabledApiPlugins',
-        message: 'Select the plugins to enable',
-        choices: Object.values(PLUGIN_NAMES)
+/**
+ * Writes given config structure into a file.
+ */
+function persistConfig(storagePath: string, config: ConfigFile) {
+    const dirPath = path.dirname(storagePath)
+
+    if (!existsSync(dirPath)) {
+        mkdirSync(dirPath, {
+            recursive: true,
+        })
     }
 
-    const portPrompts: Array<inquirer.Question> = Object.keys(DEFAULT_CONFIG_PORTS).map((key) => {
-        const name = PLUGIN_NAMES[key]
-        const defaultPort = DEFAULT_CONFIG_PORTS[key]
-        return {
-            type: 'input',
-            name: `${name}Port`,
-            message: `Provide a port for the ${name} Plugin [Enter for default: ${defaultPort}]`,
-            when: (answers: inquirer.Answers) => {
-                return answers.enabledApiPlugins.includes(name)
+    writeFileSync(storagePath, JSON.stringify(config, null, 4))
+
+    chmodSync(storagePath, '600')
+}
+
+/**
+ * Adjusts the given config for the Mumbai test environment.
+ */
+function getMumbaiConfig(config: ConfigFile): ConfigFile {
```

Comment:
We can use the new environment preset, i.e. just defined
```
    environment: 'mumbai'
```

https://github.com/streamr-dev/network/pull/2139/files

Reply:
Very handy! Let's make proper adjustments once the env feature is live.

Reply:
It will be live in the same release as this config wizard